### PR TITLE
T-14675 feat: add basic ci to portal api go

### PR DESCRIPTION
**Pull request by pablocampogo**
- Add basic CI for repo (some linting was done more loosely than in other repos for speed)
- Comment not passing tests for now (we still want to run the tests for packages that are currently being used and have passing unit tests)
- Fix linting issues

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ const (
 )
 
 type settings struct {
-	RpcUrls    []string
+	RPCURLs    []string
```
Comment (commoddity) [inline main.go:25]: Thanks for updating these var names

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      with:
+        go-version: 1.18
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Run Revive Action
+      uses: docker://morphy/revive-action:v2
+      with:
+        config: revive.toml
```
Comment (commoddity) [inline .github/workflows/go.yml]: Is Revive meant to be used on top of all the go linting tools that run as part of the precommit flow? 

Should we maybe add Revive to the pre-commit command as well so we don't end up with possibly conflicting linter rules between the CI and pre-commit hooks?

Comment (pablocampogo) [reply .github/workflows/go.yml]: Honestly I was just copyng and pasting from other repos, we can discuss this on monday, but yes we should just use one tool

Comment (commoddity) [reply .github/workflows/go.yml]: Cool SG

**Review by commoddity — APPROVED** [review]

**Pull request merged**